npms/skale-ima/index.js

```diff
@@ -137,6 +137,52 @@ async function get_contract_call_events( joContract, strEventName, nBlockNumber,
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+function compose_tx_instance( strLogPrefix, rawTx ) {
+    strLogPrefix = strLogPrefix || "";
+    rawTx = JSON.parse( JSON.stringify( rawTx ) ); // clone
+    let joOpts = null;
+    if( "chainId" in rawTx && typeof rawTx.chainId == "number" ) {
+        switch ( rawTx.chainId ) {
+        case 1:
+            delete rawTx.chainId;
+            joOpts = joOpts || { };
+            joOpts.chain = "mainnet";
+            break;
+        case 3:
+            delete rawTx.chainId;
+            joOpts = joOpts || { };
+            joOpts.chain = "ropsten";
+            break;
+        case 4:
+            delete rawTx.chainId;
+            joOpts = joOpts || { };
+            joOpts.chain = "rinkeby";
+            break;
+        case 5:
+            delete rawTx.chainId;
+            joOpts = joOpts || { };
+            joOpts.chain = "goerli";
+            break;
+        case 2018:
```

Comment:
What is the secret number 2018?:)

Reply:
**2018** is **chainId** of network called **dev**. It's listed in common set of networks with **Rinkeby** and **Ropsten**.